### ISSUE TITLE
OfflineSAM/OfflineAddAdmin2 compile

### ISSUE THREAD

**Issue by Scaum** [gtworek/PSBits#22]
Hi,

Trying to compile the OfflineAddAdmin2 but I'm having issues. Managed to compile using VS Studio with the following command:
`cl OfflineAddAdmin.c legacy_stdio_definitions.lib advapi32.lib`, however the resulting exe is not working. When I reproduce the exploit, I get the following error: `LSAOfflineOpenPolicy... error 0xc000000d`.

Using your exe works however.

My code is exactly the same as yours, I only changed the username and password. However my resulting exe is 139KB while yours is 141KB.

Would you have an idea what could be the issue ? Could you share the exact command used for compilation ?

**Comment by Scaum** [comment]
Thanks for answering.

Yes I'm doing it offline. I can use your EXE just fine and it works, but when I compile the exact same code I'm getting the error.
I would of course like to be able to compile it myself as I won't be able to use an EXE from someone else on future engagements (I do trust you, but we are never too careful ;) )

Trying to debug and it seems that the issue comes from
`return pfnLsaOfflineOpenPolicy(pszWindowsDirectory, PolicyHandle);` (on line 74).
The `GetProcAddress` call just before returns a valid handle, but if I try to log the `hPolicy` value after the call `Status = LsaOfflineOpenPolicy(wszWorkingDir, &hPolicy);` on line 406, it returns `0000000000000000`.

The call to `STATUSCHECK` on line 407 is the one printing the error and exiting the program.

Did you also compile with the same command as me ?

I'll keep trying to find a solution but would appreciate any hints if someone has an idea.

**Comment by gtworek** [comment]
I fully understand you, I would like to build on my own as well. :)
I am using Visual Studio 2019. The Community (free) edition should be ok. I can share entire solution if you need it, or we can try to debug the issue.

**Comment by gtworek** [comment]
Files added. Please let me know how it ended, it may have some value for the community.

**Comment by Scaum** [comment]
Thanks a lot dude.
I didn't even have to use VS 2019. Once I opened you SLN I had a popup saying the project was for an older version and whether I wanted to upgrade or not. After selecting "No upgrade", I was able to simply build and it worked like a charm.

Not 100% sure what was causing the issue when starting from scratch (I'm not an expert C dev) but anyway it works when loading the whole solution.

Thanks again for you reactivity, I'm closing the issue now.
